### ISSUE TITLE
Long opening and saving times when working with a big pdf file

### ISSUE THREAD

**Issue by nando-digital** [flxzt/rnote#1024]
**Describe the bug**  
When opening a big pdf (in my case a  textbook with 286 pages) rnote needs a long time (27 seconds) to open and (57 seconds) to save the document. 

**To Reproduce**  
1. Open a rnote file containing a big pdf.
2. Wait.
3. Save the file.
4. Wait.

**Expected behavior**  
Way smaller loading times. Rnote would be useful for having a textbook with exercises and solving them right next to the pdf, but it's not ideal to have to wait such a long time to open the file.
I don't know how rnote works on the inside, but my guess is that the whole document is loaded, instead of just the relevant part and the whole document is saved instead of just the changes. 

**Desktop (please complete the following information):**  
 - OS: Arch Linux
 - App Version: Rnote v0.9.4
 - Installation Source: Archlinux Extra Repo
 - Desktop Environment: Hyprland
 - Display Server: Wayland
 - Input Source: Convertible laptop (mouse, keyboard, touchscreen, pen)

**Comment by Doublonmousse** [comment]
Yes, rnote loads the whole pdf and converts it either into bitmap or vector images that are then saved in the rnote file.

I don't know if there's any obvious performance improvement possible (like multithreading) to load pdfs faster.

Relevant discussion for a pdf annotation mode #153

**Comment by NXTler** [comment]
From all I can tell, PDFs seem to be converted into vector images with a single thread. Perhaps using one thread per page would be possible instead?
